source/dbo/Stored Procedures/Data Migration/uspMigrationOutcomeMismatch.sql

```diff
@@ -0,0 +1,38 @@
+﻿/*
+	Note that we treat some outcomes as being equal even when they are actually different in NTBS and ETS.
+	For more details see uspMigrationMisMatchedOutcomes.sql.
+*/
+
+CREATE PROCEDURE [dbo].[uspMigrationOutcomeMismatch]
+	@MigrationRun INT = NULL
+AS
+	WITH EventOrder AS
+	(SELECT 'DiagnosisMade' AS EventName, 1 AS 'OrderBy'
+		UNION
+		SELECT 'TreatmentStart' AS EventName, 2 AS 'OrderBy'
+		UNION
+		SELECT 'TransferOut' AS EventName, 3 AS 'OrderBy'
+		UNION
+		SELECT 'TransferIn' AS EventName, 4 AS 'OrderBy'
+		UNION
+		SELECT 'TransferRestart' AS Eventname, 5 AS 'OrderBy'
+		UNION
+		SELECT 'TreatmentOutcome' AS EventName, 6 AS 'OrderBy')
+
+	SELECT DISTINCT
+		mrr.MigrationNotificationId		AS 'MigrationNotificationId',
+		mrr.LegacyETSId					AS 'EtsId',
+		mrr.NTBSNotificationId			AS 'NtbsId',
+		mrr.EtsTreatmentOutcome			AS 'EtsOutcome',
+		mrr.NTBSTreatmentOutcome		AS 'NtbsOutcome',
+		FIRST_VALUE(te.EventDate) OVER (PARTITION BY NotificationId ORDER BY EventDate DESC, OrderBy DESC)			AS 'FinalEventDate',
+		FIRST_VALUE(te.TreatmentEventType) OVER (PARTITION BY NotificationId ORDER BY EventDate DESC, OrderBy DESC)	AS 'FinalEventType'
+	FROM [dbo].[MigrationRunResults] mrr
+		INNER JOIN [$(NTBS)].[dbo].[TreatmentEvent] te ON te.NotificationId = mrr.NTBSNotificationId
```

Comment:
I've used an inner join here, because I think every notification is guaranteed to have at least one treatment event, but looking at this now, perhaps a left join would have been safer

Reply:
Yeah I would go for a left join just to be safe. An inner join _should_ never fail here but you never know.